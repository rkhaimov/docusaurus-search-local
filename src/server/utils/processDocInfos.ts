import path from "path";
import { DocInfoWithFilePath, PostBuildData } from "../../shared/interfaces";

export function processDocInfos({
  routesPaths,
  outDir,
  baseUrl,
}: PostBuildData) {
  return routesPaths.reduce((accumulated, url) => {
    const route = url.substr(baseUrl.length);

    if (route === "404.html") {
      return accumulated;
    }

    accumulated.push({
      filePath: path.join(outDir, route, "index.html"),
      url,
      type: "docs",
    });

    return accumulated;
  }, [] as DocInfoWithFilePath[]);
}
