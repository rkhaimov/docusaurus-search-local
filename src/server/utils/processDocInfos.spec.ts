import { PostBuildData } from "../../shared/interfaces";
import { processDocInfos } from "./processDocInfos";

describe("processDocInfos", () => {
  const routesPaths: string[] = [
    "/",
    "/introduction",
    "/pluralize",
    "/404.html",
  ];

  const buildData: PostBuildData = {
    routesPaths,
    outDir: "/build",
    baseUrl: "/",
  };

  test("processDocInfos works for rootDocs", () => {
    expect(processDocInfos(buildData)).toMatchSnapshot();
  });
});
