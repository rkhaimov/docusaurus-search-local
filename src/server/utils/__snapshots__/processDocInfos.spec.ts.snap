// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`processDocInfos processDocInfos works for rootDocs 1`] = `
Array [
  Object {
    "filePath": "\\\\build\\\\index.html",
    "type": "docs",
    "url": "/",
  },
  Object {
    "filePath": "\\\\build\\\\introduction\\\\index.html",
    "type": "docs",
    "url": "/introduction",
  },
  Object {
    "filePath": "\\\\build\\\\pluralize\\\\index.html",
    "type": "docs",
    "url": "/pluralize",
  },
]
`;
